src/microbe_stage/editor/CellEditorComponent.Callbacks.cs

```diff
@@ -308,8 +309,9 @@ private void UndoOrganelleUpgradeAction(OrganelleUpgradeActionData data)
     {
         data.UpgradedOrganelle.Upgrades = data.OldUpgrades;
 
-        // Uncomment when upgrades can visually affect the cell
-        // UpdateAlreadyPlacedVisuals();
+        microbeVisualizationOrganellePositionsAreDirty = true;
+
+        OnOrganellesChanged();
 
         UpdateStats();
```

Comment:
Also the `UpdateStats` call should be removed from here and the do method as organelles changed already does that.